File: Python/Structures/SinglyList.py
class Node:
    def __init__(self, data=None) -> None:
        self.data = data
        self.next = None


class SinglyList:
    def __init__(self) -> None:
        self.head = None

    def add(self, obj: int) -> None:
        node = Node(obj)
        node.next = self.head
        self.head = node

    def display(self) -> None:
        ptr = self.head
        while ptr.next != None:
            print(ptr.data)
            ptr = ptr.next

    def is_empty(self) -> bool:
        pass

    def remove(self, obj: int) -> None:
        pass


if __name__ == '__main__':
    s = SinglyList()
    for i in range(0, 9):
        s.add(i)
    s.display()

File: Dart/Structures/SinglyList.dart
class Node {
  int data = 0;
  Node? next;

  Node(int data) {
    this.data = data;
    this.next = null;
  }
}

class SinglyList {
  Node? head;
  late int size;

  SinglyList() {
    this.head = null;
    this.size = 0;
  }

  void add(int data) {
    Node node = new Node(data);
    if (isEmpty()) {
      this.head = node;
      ++this.size;
    } else {
      Node? ptr = this.head;
      while (ptr?.next != null) ptr = ptr?.next;

      ptr?.next = node;
      ++this.size;
    }
  }

  void remove(int data) {
    Node? ptr = this.head;
    Node? prev = ptr;

    while (ptr?.next != null) {
      if (ptr?.data == data) {
        prev?.next = ptr?.next;
        --this.size;
        return;
      }
      prev = ptr;
      ptr = ptr?.next;
    }
  }

  void display() {
    Node? ptr = this.head;
    while (ptr?.next != null) {
      print("${ptr?.data} ");
      ptr = ptr?.next;
    }
  }

  bool isEmpty() {
    return this.head == null;
  }
}

void main() {
  var l = new SinglyList();
  for (int i = 0; i < 10; ++i) l.add(i);

  l.display();
  l.remove(5);
  print("5 is removed.");
  l.display();
}
